src/util/amount.rs

```diff
@@ -67,16 +67,47 @@ impl FromStr for Denomination {
     type Err = ParseAmountError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "BTC" => Ok(Denomination::Bitcoin),
-            "mBTC" => Ok(Denomination::MilliBitcoin),
-            "uBTC" => Ok(Denomination::MicroBitcoin),
-            "bits" => Ok(Denomination::Bit),
-            "satoshi" => Ok(Denomination::Satoshi),
-            "sat" => Ok(Denomination::Satoshi),
-            "msat" => Ok(Denomination::MilliSatoshi),
-            d => Err(ParseAmountError::UnknownDenomination(d.to_owned())),
+        if s == "Msat" {        // So as not to confuse with mega sats.
+            return Err(ParseAmountError::UnknownDenomination(s.to_owned()));
         }
+        if s == "Msats" {
+            return Err(ParseAmountError::UnknownDenomination(s.to_owned()));
+        }
+
+        if s.eq_ignore_ascii_case("BTC") {
+            return Ok(Denomination::Bitcoin)
+        }
+        if s.eq_ignore_ascii_case("mBTC") {
+            return Ok(Denomination::MilliBitcoin);
+        }
+        if s.eq_ignore_ascii_case("uBTC") {
+            return Ok(Denomination::MicroBitcoin);
+        }
```

Comment:
Ha ha, I wanted to do this as well but remember we cannot index strings in Rust because of utf8. I didn't want to do an allocation so no `String` methods either. Then I tried `strip_suffix`, not noticing it was Rust 1.45. Hence I reverted to the ugly if clauses again.

Reply:
> we cannot index strings in Rust because of utf8

We can, we just have to do it by bytes and given that `s` and `S` are ASCII, it's exactly 1 byte, so indexing is fine (and if we're lucky the compiler will even optimize-out bounds check).

Reply:
Good points!

Please re-review, its looking hot now :)